include/dqlite.h

```diff
@@ -399,16 +413,6 @@ DQLITE_API int dqlite_node_set_snapshot_params(dqlite_node *n,
  */
 DQLITE_API int dqlite_node_set_block_size(dqlite_node *n, size_t size);
 
-/**
- * WARNING: This is an experimental API.
- *
- * By default dqlite holds the SQLite database file and WAL in memory. By
- * enabling disk-mode, dqlite will hold the SQLite database file on-disk while
- * keeping the WAL in memory. Has to be called after `dqlite_node_create` and
- * before `dqlite_node_start`.
- */
-DQLITE_API int dqlite_node_enable_disk_mode(dqlite_node *n);
```

Comment:
Why are we changing the public API? Not that we should not, I am just wondering what are the reasons for defining `dqlite_node_create_v2` now when the feature is not yet stable.

Reply:
Basically I decided you were right in this comment: https://github.com/canonical/dqlite/pull/661#discussion_r1663785674; it seems lower-cost and lower-risk to introduce a new constructor that allows asking for disk mode up front than to make the changes to raft initialization that were required to keep dqlite_enable_disk_mode working.

Reply:
Are we going to support both types of disk_mode in the future? The only thing I am worried about is having to introduce `_v3` version of the function soon.

Reply:
No, the old disk mode is going away.